airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_ads_insights_action_product_id.py

```diff
@@ -0,0 +1,489 @@
+        job_ids = [job_ids]
+    body = [
+        {
+            "body": json.dumps(
+                {
+                    "id": job_id, "account_id": account_id, "async_status": status, "async_percent_completion": 100
+                }
+            ),
+        } for job_id in job_ids
+    ]
+    return build_response(body=body, status_code=HTTPStatus.OK)
+
+
+def _insights_response() -> HttpResponseBuilder:
+    return create_response_builder(
+        response_template=find_template(_STREAM_NAME, __file__),
+        records_path=FieldPath("data"),
+        pagination_strategy=FacebookMarketingPaginationStrategy(_get_insights_request(_JOB_ID).with_limit(100).build()),
+    )
+
+
+def _ads_insights_action_product_id_record() -> RecordBuilder:
+    return create_record_builder(
+        response_template=find_template(_STREAM_NAME, __file__),
+        records_path=FieldPath("data"),
+        record_cursor_path=FieldPath(_CURSOR_FIELD),
+    )
+
+
+@freezegun.freeze_time(NOW.isoformat())
+class TestFullRefresh(TestCase):
+
+    @staticmethod
+    def _read(config_: ConfigBuilder, expecting_exception: bool = False) -> EntrypointOutput:
+        return read_output(
+            config_builder=config_,
+            stream_name=_STREAM_NAME,
+            sync_mode=SyncMode.full_refresh,
+            expecting_exception=expecting_exception,
+        )
+
+    @HttpMocker()
+    def test_given_one_page_when_read_then_return_records(self, http_mocker: HttpMocker) -> None:
+        http_mocker.get(get_account_request().build(), get_account_response())
+        http_mocker.get(_update_api_throttle_limit_request().build(), _update_api_throttle_limit_response())
+        http_mocker.post(_job_start_request().build(), _job_start_response(_REPORT_RUN_ID))
+        http_mocker.post(_job_status_request(_REPORT_RUN_ID).build(), _job_status_response(_JOB_ID))
+        http_mocker.get(
+            _get_insights_request(_JOB_ID).build(),
+            _insights_response().with_record(_ads_insights_action_product_id_record()).build(),
+        )
+
+        output = self._read(config())
+        assert len(output.records) == 1
+
+    @HttpMocker()
+    def test_given_multiple_pages_when_read_then_return_records(self, http_mocker: HttpMocker) -> None:
+        http_mocker.get(get_account_request().build(), get_account_response())
+        http_mocker.get(_update_api_throttle_limit_request().build(), _update_api_throttle_limit_response())
+        http_mocker.post(_job_start_request().build(), _job_start_response(_REPORT_RUN_ID))
+        http_mocker.post(_job_status_request(_REPORT_RUN_ID).build(), _job_status_response(_JOB_ID))
+        http_mocker.get(
+            _get_insights_request(_JOB_ID).build(),
+            _insights_response().with_pagination().with_record(_ads_insights_action_product_id_record()).build(),
+        )
+        http_mocker.get(
+            _get_insights_request(_JOB_ID).with_pagination_parameter().build(),
```

Comment:
I think it would be nice to see what is this parameter and where it comes from. The same applies to test_videos

Reply:
Thanks for the advice, made the following updates:

- Renamed `with_pagination_parameter` to `with_next_page_token`
- Added required `next_page_token: str` parameter to the `with_next_page_token` method so we can see which token is used
- Added required `next_page_token: str` parameter to the `FacebookMarketingPaginationStrategy.__init__` method so we can control which next page token is used at the test module level